boards/Makefile.common

```diff
@@ -43,35 +43,6 @@ endif
 # cargo `--target_dir`.
 TARGET_DIRECTORY ?= $(TOCK_ROOT_DIRECTORY)target/
 
-# RUSTC_FLAGS allows boards to define board-specific options.
-# This will hopefully move into Cargo.toml (or Cargo.toml.local) eventually.
-#
-# - `-Tlayout.ld`: Use the linker script `layout.ld` all boards must provide.
-# - `linker=rust-lld`: Tell rustc to use the LLVM linker. This avoids needing
-#   GCC as a dependency to build the kernel.
-# - `linker-flavor=ld.lld`: Use the LLVM lld executable with the `-flavor gnu`
-#   flag.
-# - `relocation-model=static`: See https://github.com/tock/tock/pull/2853
-# - `-nmagic`: lld by default uses a default page size to align program
-#   sections. Tock expects that program sections are set back-to-back. `-nmagic`
-#   instructs the linker to not page-align sections.
-# - `-icf=all`: Identical Code Folding (ICF) set to all. This tells the linker
-#   to be more aggressive about removing duplicate code. The default is `safe`,
-#   and the downside to `all` is that different functions in the code can end up
-#   with the same address in the binary. However, it can save a fair bit of code
-#   size.
-# - `-C symbol-mangling-version=v0`: Opt-in to Rust v0 symbol mangling scheme.
-#   See https://github.com/rust-lang/rust/issues/60705 and
-#   https://github.com/tock/tock/issues/3529.
-RUSTC_FLAGS ?= \
-  -C link-arg=-Tlayout.ld \
-  -C linker=rust-lld \
-  -C linker-flavor=ld.lld \
-  -C relocation-model=static \
-  -C link-arg=-nmagic \
-  -C link-arg=-icf=all \
-  -C symbol-mangling-version=v0 \
-
 # RISC-V-specific flags.
 ifneq ($(findstring riscv32i, $(TARGET)),)
   # NOTE: This flag causes kernel panics on some ARM cores. Since the size
```

Comment:
How do you add these flags? Every board has to remember to do it?

Reply:
These reside in `boards/.cargo/config.toml`, which gets merged into all board crates' config (layered configuration system): https://github.com/tock/tock/blob/c056bec40da4bb97ad498248bc5b83896b51a710/boards/.cargo/config.toml

Reply:
These are?

```
  # benefit is almost exclusively for RISC-V, we only apply it for those
  # targets.
  RUSTC_FLAGS += -C force-frame-pointers=no
  # Ensure relocations generated is eligible for linker relaxation.
  # This provide huge space savings.
  RUSTC_FLAGS += -C target-feature=+relax
```

Reply:
The RISC-V specific ones are added in RISC-V specific board `.cargo/config.toml`s

Reply:
Where?

Reply:
@bradjc I believe you are right, they seem to not be passed right now. I didn't realize you were referring to the RISC-V flags, and not the generic ones above. Should be easy to add to all RISC-V boards.

Reply:
Github says no commenting on unmodified lines!